src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const sexFiltered = people.filter(peopleInfo => peopleInfo.sex === 'm');
+  const sexAndCenturyFiltered = sexFiltered.filter(
+    peopleInfo => Math.ceil(peopleInfo.died / 100) === century
+  );
+
+  const filtered = century
+    ? sexAndCenturyFiltered
+    : sexFiltered;
+
+  return (filtered.reduce((sum, peopleInfo) => (
+    sum + (peopleInfo.died - peopleInfo.born)
+  ), 0)) / filtered.length;
```

Comment:
move this to the separate function & reuse it
```suggestion
  return getAverageAge(filtered);
```